_posts/2025-08-28-quarkus-a2a-java-0-3-0-alpha-release.adoc

```diff
@@ -0,0 +1,242 @@
+<4> The primary endpoint URL should also be specified here for completeness.
+<5> The gRPC transport URL, `localhost:9000`, indicating that our A2A server agent also supports gRPC.
+
+=== A2A Client Updates
+
+To make use of an A2A client using the A2A Java SDK, add the following dependency to your project:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.github.a2asdk</groupId>
+    <artifactId>a2a-java-sdk-client</artifactId> <1>
+</dependency>
+----
+<1> The `a2a-java-sdk-client` dependency provides access to a `ClientBuilder` that you can use to create your A2A `Client`.
+
+==== Supported Transports
+
+The A2A Java SDK client implementation now supports the gRPC transport protocol in addition to JSON-RPC.
+
+As mentioned previously, support for the HTTP+JSON/REST transport will be coming soon!
+
+===== JSON-RPC 2.0 Transport
+
+To allow your client to support the JSON-RPC transport for communication, just `a2a-java-sdk-client` is needed. No additional dependencies need to be manually added to your project.
+
+===== gRPC Transport
+
+To allow your client to support the gRPC transport for communication, simply add the following dependency to your project:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.github.a2asdk</groupId>
+    <artifactId>a2a-java-sdk-client-transport-grpc</artifactId>
+</dependency>
+----
+
+==== Creating a `Client`
+
+An A2A `Client` can now be created using a `ClientBuilder`. The `ClientBuilder` will select the appropriate transport protocol to use based on information obtained from the `AgentCard` for the A2A server agent that this client will be communicating with, also taking into account client-specified transport configuration. The `ClientBuilder` will then create a `Client` that will use the selected transport to communicate with the A2A server and the `Client` will also make use of any other client-specified configuration like whether to use streaming (`true` by default), accepted output modes, history length for messages, and so on.
+
+We want to give a shoutout to David Brassely, one of our community contributors, for working with us on improving the `Client` creation experience!
+
+Let's take a look at an example to see how a `Client` can now be created using the `ClientBuilder`:
+
+[source,java]
+----
+// First, get the agent card for the A2A server agent you want to connect to
+AgentCard agentCard = new A2ACardResolver("http://localhost:1234").getAgentCard();
+
+// Specify general client configuration and preferences for the ClientBuilder
+ClientConfig clientConfig = new ClientConfig.Builder()
+        .setAcceptedOutputModes(List.of("text"))
+        ...
+        .build();
+
+// Create event consumers to handle responses that will be received from the A2A server
+// (these consumers will be used for both streaming and non-streaming responses)
+List<BiConsumer<ClientEvent, AgentCard>> consumers = List.of(
```

Comment:
A single stream with a consumer using pattern matching may provide a slightly more elegant (to be checked) API.